src/utils/splits.ts

```diff
@@ -87,26 +89,96 @@ export const sortSplits = (splits: Split[]) => {
   return [...splits].sort((a, b) => (a.percent < b.percent ? 1 : -1))
 }
 
-// returns all unique splits (projectIds or addresses), sorted by their new `percent`
+/* Determines if two splits AMOUNTS are equal. Extracts amounts for two splits from their respective totalValues **/
+export function splitAmountsAreEqual({
+  split1,
+  split2,
+  split1TotalValue,
+  split2TotalValue,
+}: {
+  split1: Split
+  split2: Split
+  split1TotalValue?: BigNumber
+  split2TotalValue?: BigNumber
+}) {
+  const split1Amount = formatWad(
+    split1TotalValue?.mul(split1.percent).div(SPLITS_TOTAL_PERCENT),
+    {
+      precision: 2,
+    },
+  )
+  const split2Amount = formatWad(
+    split2TotalValue?.mul(split2.percent).div(SPLITS_TOTAL_PERCENT),
+    {
+      precision: 2,
+    },
+  )
+  return split2Amount === split1Amount
+}
+
+/* Determines if two splits are equal. If given totalValues, uses the amount of each split instead of its percent **/
+function splitsAreEqual({
+  split1,
+  split2,
+  split1TotalValue,
+  split2TotalValue,
+}: {
+  split1: Split
+  split2: Split
+  split1TotalValue?: BigNumber
+  split2TotalValue?: BigNumber
+}) {
+  const isFiniteTotalValue =
+    isFiniteDistributionLimit(split1TotalValue) &&
+    isFiniteDistributionLimit(split2TotalValue)
+  if (!isFiniteTotalValue) {
+    return isEqual(split1, split2)
+  }
+  return (
+    splitAmountsAreEqual({
+      split1,
+      split2,
+      split1TotalValue,
+      split2TotalValue,
+    }) &&
+    split1.beneficiary === split2.beneficiary &&
+    split1.allocator === split2.allocator &&
+    split1.lockedUntil === split2.lockedUntil &&
+    split1.projectId === split2.projectId &&
+    split1.preferClaimed === split2.preferClaimed
```

Comment:
**suggestion (non-blocking)** pretty sure there's a lodash object deep equal checker util. might be worth looking into, to save this bongery